### PR TITLE
[EDIFICE] Badge components

### PR DESCRIPTION
**Pull request by jcbe-ode**
# Description

Implementation of a a generic badge component in React : 
* relies on bootstrap Badge,
* declares 3 variants : Notification / Profile / Link,
* each variant having some additional parameters.

Also, this PR allows editing an internal link (which is rendered as a Badge) by opening the MediaLibrary.
Some imperative handlers have been added for this to work.

## Which Package changed?

Please check the name of the package you changed

- [X] Components
- [ ] Core
- [ ] Icons
- [ ] Hooks
- [X] Multimedia

## Has the documentation changed?

- [X] Storybook

## Type of change

Please check options that are relevant.

- [ ] Chore (PATCH)
- [ ] Doc (PATCH)
- [ ] Bug fix (PATCH)
- [ ] New feature (MINOR)
- [X] Breaking change (MAJOR)

# Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings

### CODE REVIEW

File: packages/react/src/components/Modal/Modal.tsx
Comment (ccreusat) [inline packages/react/src/components/Modal/Modal.tsx:1]: @jcbe-ode pour quoi modale à changer ? juste `forwardRef((props: ModalProps, ref: Ref<ModalElement>) => {...})` ?

File: packages/react/src/multimedia/Linker/InternalLinker.tsx [hunk 1]
```diff
@@ -108,6 +114,43 @@ const InternalLinker = ({
...
+    setSelectedApplication(option);
+  };
+
+  // Handle search input events (and debounce)
+  const handleSearchChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const newText = event.target.value;
+    setSearchTerms(newText.toString());
+  };
+  const handleSubmit = useCallback(
+    (e: FormEvent) => {
```
Comment (ccreusat) [inline packages/react/src/multimedia/Linker/InternalLinker.tsx:129]: @jcbe-ode explicit code > e = event 🙏🏼

File: packages/react/src/multimedia/Linker/InternalLinker.tsx [hunk 2]
```diff
@@ -108,6 +114,43 @@ const InternalLinker = ({
...
+      e.preventDefault();
+    },
+    [loadAndDisplayResources, searchTerms],
+  );
+
+  // Handle [de-]selection of a resource by the user.
+  const toggleResourceSelection = useCallback(
+    (resource: ILinkedResource) => {
+      const index = selectedDocuments.findIndex(
+        (doc) => doc.assetId === resource.assetId,
```
Comment (ccreusat) [inline packages/react/src/multimedia/Linker/InternalLinker.tsx:141]: @jcbe-ode là je chipote aussi, mais si le tableau c'est `selectedDocuments` , l'élément devrait être `selectedDocument`

**Review by ccreusat — COMMENTED** [review]
A part les petits changements et la question, m'a l'air ok 👍🏼

File: packages/react/src/components/Modal/Modal.tsx
Comment (jcbe-ode) [reply packages/react/src/components/Modal/Modal.tsx:1]: Oui c'est uniquement le typage qui a été clarifié : 
* ajout du type ModalElement,
* passage de `forwardRef<HTMLDivElement, ModalProps>` à `forwardRef((props: ModalProps, ref: Ref<ModalElement>) => `

Mais en vrai, je préfèrerais presque `forwardRef<ModalElement, ModalProps>` ... ça parait plus facile à relire.

**Pull request merged**